tests/test.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import re
+import json
+
+scriptdir = os.path.dirname(os.path.realpath(__file__))
+needledir = os.path.join(scriptdir, "..")
+
+returncode = 0
+
+def error(msg):
+  global returncode
+  returncode = 1
+  print(msg, file=sys.stderr)
+
+# Get all needles
+needles = set()
+for f in os.listdir(needledir):
+  if f.endswith('.json') or f.endswith('.png'):
+    needle, ext = f.rsplit('.', 1)
+    needles.add(needle)
+
+# Check for missing parts
+for needle in sorted(needles):
+  jsonfile = os.path.join(needledir, needle + '.json')
+  pngfile = os.path.join(needledir, needle + '.png')
+
+  # Check for file existence
+  if not os.path.isfile(jsonfile):
+    error("Needle '{}' is missing its JSON file!".format(needle))
+    continue # parsing the json makes no sense
+
+  if not os.path.isfile(pngfile):
+    error("Needle '{}' is missing its PNG file!".format(needle))
+
+  # Check JSON content
+  n = None
+  with open(jsonfile) as f:
+    n = json.load(f)
+
+  # Check if workaround tag exists if bugref is in name
+  if re.search(r'-(poo|bsc|bnc|boo)[0-9]+-', needle):
+    if not 'workaround' in n['properties']:
+      error("Needle '{}' has bugid in name but no workaround tag set!".format(needle))
```

Comment:
Actually I would allow to mention a bug in the name without a workaround property. Might be a "RESOLVED WONTFIX" bug for reference. However, we should disallow the other way around, workaround property must always be accompanied by a bugref in the name.

Reply:
Something like this?

```diff
diff --git a/tests/test.py b/tests/test.py
index 518449b5..28129658 100755
--- a/tests/test.py
+++ b/tests/test.py
@@ -41,8 +41,10 @@ for needle in sorted(needles):
     n = json.load(f)
 
   # Check if workaround tag exists if bugref is in name
-  if re.search(r'-(poo|bsc|bnc|boo)[0-9]+-', needle):
-    if not 'workaround' in n['properties']:
-      error("Needle '{}' has bugid in name but no workaround tag set!".format(needle))
+  needle_has_bugref = bool(re.search(r'-(poo|bsc|bnc|boo)[0-9]+-', needle))
+  needle_has_workaround = 'workaround' in n.get('properties', [])
+
+  if needle_has_workaround and not needle_has_bugref:
+    error("Needle '{}' has workaround tag but no bugid in name!".format(needle))
 
 sys.exit(returncode)
```